controllers/baseboardmanagement_controller.go

```diff
@@ -119,7 +122,7 @@ func (r *BaseboardManagementReconciler) reconcile(ctx context.Context, bm *bmcv1
 	bmcClient, err := r.bmcClientFactory(ctx, bm.Spec.Connection.Host, strconv.Itoa(bm.Spec.Connection.Port), username, password)
 	if err != nil {
 		logger.Error(err, "BMC connection failed", "host", bm.Spec.Connection.Host)
-		result, setConditionErr := r.setCondition(ctx, bm, bmPatch, bmcv1alpha1.ConnectionError, err.Error())
+		result, setConditionErr := r.setCondition(ctx, bm, bmPatch, bmcv1alpha1.Connected, bmcv1alpha1.BaseboardManagementConditionFalse, err.Error())
```

Comment:
Could we break this apart into 2 calls rather than bundling into 1 so it only does 1 thing and is consistent with reconciliation funcs. Right now it seems we're jumping between levels of abstraction.

```go
if err != nil {
  // Mutate the BaseboardManagement type.
  r.setCondition(bm, bmcv1alpha1.Connected, bmcv1alpha1.BaseboardManagementConditionFalse, err.Error())

  // Apply the patch separately.
  if err := r.applyPatch(ctx, bmPatch, bm); err != nil {
     // Handle
  }
}
```

Its probably useful to define the `setCondition` on the `BaseboardManagement` type as a helper method also and just call that.

```go
bm.SetCondition(bmc.v1alpha1.Connected, bmcv1alpha1.BaseboardManadementConditionFalse, err.Error())
```

And if we go the full mile, this method should have a variadic option set because the `message` is optional and we're forcing consumers to specify `""` at the end if no message is required, which is nasty. Its a little more code, but a better API.

```go
// api/v1alpha1 pkg

type BaseboardManagementSetConditionOption func(*BaseboardManagementCondition)

func (bmc BaseboardManagement) SetCondition(condition BaseboardManagementConditionType, status BaseboardManagementConditionStatus, opt BaseboardManagementSetConditionOption) error {
  ...
}

func WithConditionMessage(m string) BaseboardManagementSetConditionOption {
  return func(c *BaseboardManagementCondition) {
    c.Message = m
  }
}
```

Reply:
Sounds good 👍  I like the idea of having `SetCondition` on the types itself and having the options.
Will follow a similar model for all the types, it makes it simpler.